### ISSUE TITLE
Link encoding bug in some widgets

### ISSUE THREAD

**Issue by ggeisler** [projectblacklight/spotlight#2160]
In some of the widgets that offer a text field, if you create a link in the text that includes certain Markdown formatting characters, the link URL will be incorrectly encoded.

As best I can tell, the characters that cause the problem are:

- single underscore: `_sometext_` becomes `<i>sometext</i>`
- double underscore: `__sometext__` `<i></i>sometext<i></i>`
- double asterisk: `**sometext**` `<b>sometext</b>`

(Surprisingly, a single asterisk doesn't seem to be a problem.)

The widgets affected are:

- Embed + Text
- Uploaded Item Row
- Item Row
- Item Embed
- Item Grid

An odd thing that makes it a bit tricky to test is that the bug doesn't show up the first time the page with a widget containing a link with the problem characters is saved. But it reliably occurs if you re-edit the page once more, and every time you edit the page after that. To reproduce:

1. Create or edit an existing feature page.
2. Add an Embed + Text widget, or any from the list above.
3. In the Text field, add some text.
4. Highlight some of the text and click the link icon from the formatting tools menu to add a link.
5. Add a URL that includes an single or double underscore or double asterisk (e.g., `http://example.com/foo_bar_`
6. Save the page

If you check the link it might look fine the first time, but edit the page again and re-save (you don't need to actually make any change to the page, just save it).  The link should now look like `http://example.com/foo<i>bar</i>`

**Comment by mejackreed** [comment]
This seems to be known upstream in https://github.com/madebymany/sir-trevor-js/issues/232 with a fix in https://github.com/madebymany/sir-trevor-js/pull/274 . The fix is to store HTML rather than markdown. Investigating possible solutions.

**Comment by cbeer** [comment]
Fixed by 06bb56c392bef155956f89409c94e08db8cd17ad

**Comment by mejackreed** [comment]
A handy way to remediate content:

```ruby
Spotlight::Page.where('content LIKE ?', "%&lt;i&gt;%").find_each { |p| p.update_column(:content, p[:content].gsub(/&lt;\/?i&gt;/, '_')) }
```
